src/gridfs/download.ts

```diff
@@ -183,24 +183,13 @@ function doRead(stream: GridFSBucketReadStream): void {
   if (!stream.s.cursor) return;
   if (!stream.s.file) return;
 
-  const handleReadResult = ({
-    error,
-    doc
-  }: { error: Error; doc: null } | { error: null; doc: any }) => {
-    if (stream.destroyed) {
-      return;
-    }
-    if (error) {
-      stream.destroy(error);
-      return;
-    }
+  const handleReadResult = (doc: Document | null) => {
```

Comment:
Moved the error case to the rejection handler since it was a one-liner. It will make a future conversion (if ever) to async easier